### ISSUE TITLE
How does Ginga handle image distortion?

### ISSUE THREAD

**Issue by pllim** [ejeschke/ginga#328]
Is there a way to un-distort a distorted image (using its WCS) and display the un-distorted version on Ginga? This was also asked by @mcara.

**Comment by pllim** [comment]
c/c @stscieisenhamer in case he secretly implemented this for 3D printing project...

**Comment by mcara** [comment]
I think this would be a great feature. I do not think DS9 is capable of doing this either.

**Comment by ejeschke** [comment]
I think if you are using `astropy` wcs there is *some* support for distortion correction, no?  And I think that even some images that have distortion information in auxiliary tables in the header can be supported.  But this probably just means that the coordinate reporting and graphing should be correct.  There is no actual warping of the image.  I wonder if that would be possible to do fairly straightforwardly using the astropy `reproject` module?

**Comment by pllim** [comment]
Yes, `reproject` was briefly discussed. The solution could be to use that to create a "corrected" image in memory and display it, and then use `SaveImage` to save it out if desired. Or to modify `Mosaic` to work with `reproject`. Of course, solution like this is prone to memory being all used up, so it is not perfect.

**Comment by ejeschke** [comment]
@pllim, revisiting this issue.  Do you have some examples of FITS files with significant distortion and screenshots of corrected images?  Would like to try out some support for correction in a branch I am working on.

**Comment by pllim** [comment]
How about ACS on HST? http://www.stsci.edu/hst/acs/analysis/distortion

You can request all its images at http://archive.stsci.edu/hst/search.php . The Wide Field Camera in ACS is what you want.

**Comment by pllim** [comment]
@ejeschke , are you still working on the support or should I just close this?

**Comment by pllim** [comment]
This came up again... cc @eteq , @hcferguson , and @larrybradley

**Comment by ejeschke** [comment]
> Yes, reproject was briefly discussed. The solution could be to use that to create a "corrected" image in memory and display it ...

I think this would be rather trivial to implement from a working example of morphing one `ndarray` to a corrected one.  If the resulting image is not rectangular, has holes, etc. this could be handled by either `NaN`s, numpy masks or an alpha mask.  The other thing that would probably need to be done is to change the WCS so that overplotting and coordinate readout under the cursor would work correctly--once you correct the image the "correction" has to be taken out of the WCS.

**Comment by ejeschke** [comment]
See #813 for a strawman approach

**Comment by ejeschke** [comment]
> Is there a way to un-distort a distorted image (using its WCS) and display the un-distorted version on Ginga?

So, with the strawman plugin, can you just construct a WCS with the distortion components removed, and use that as the reproject WCS?
